src/map/script.c

```diff
@@ -16522,6 +16522,48 @@ static BUILDIN(autoequip)
 	return true;
 }
 
+/**
+ * equipidx(<index>)
+ */
+static BUILDIN(equipidx)
+{
+	struct map_session_data *sd = script->rid2sd(st);
+
+	if (sd == NULL) {
+		script_pushint(st, 0);
+		return true;
+	}
+
+	int i = script_getnum(st, 2);
+	if (i < 0 || i >= sd->status.inventorySize) {
+		ShowError("buildin_equipidx: Index (%d) should be from 0-%d.\n", i, sd->status.inventorySize - 1);
+		script_pushint(st, 0);
+		return false;
+	}
+
+	if (sd->status.inventory[i].equip != 0) { // item already equipped, run silently
+		script_pushint(st, 1);
+		return true;
+	}
+
+	int nameid = sd->status.inventory[i].nameid;
+	struct item_data *item_data = itemdb->exists(nameid);
+	if (item_data == NULL) {
+		ShowError("buildin_equipidx: Invalid Item ID (%d).\n", nameid);
+		script_pushint(st, 0);
+		return false;
+	}
+
+	if (pc->equipitem(sd, i, item_data->equip) == 0) {
+		ShowWarning("buildin_equipidx: Item ID (%d) at index (%d) cannot be equipped.\n", nameid, i);
+		script_pushint(st, 0);
+		return true;
```

Comment:
wrong, `return true;` or `return false;` is almost the same thing
`return false;` is equal to `script->reportsrc(st);` + `return true;`
read this PR https://github.com/HerculesWS/Hercules/pull/868

what you are doing here is put up a ShowWarning without telling where this warning came from
should be `return false;` here

============== OR =================

suppress this warning, since we don't have a script command to check either this job can equip this item or not

Reply:
so to be clear, it will not stop execution of script correct